X-CASH-official/desktop-wallet
# Support 2.0.1 xcash core for MacOS and fix issues related to cli binaries download

# Description

The PR fixes the issue when the Mac OS build was malfunctioning on some of the computers. The reason was that when a packaged app installed and run for the first time it downloads binaries to the same directory where the GUI wallet is run. Sometimes it's not possible to execute a binary due to access rights.


The fix:
- when GUI starts for the first time, it creates `${HOME}/xcash-official` and `${HOME}/xcash-official/downloads` folder;
- it downloads and unpacks binaries to `${HOME}/xcash-official/downloads`;
- moves `xcash-wallet-rpc` to `${HOME}/xcash-official` and removes `downloads` directory;
- grant the access rights needed;
- run the rest of GUI wallet services as usual.

The PR also adds support for the latest XCASH Core 2.0.1 CLI binaries usage.

<img width="1552" alt="Screenshot 2021-03-29 at 8 34 05" src="https://user-images.githubusercontent.com/7963446/112792532-544afb80-906c-11eb-9ac2-28e9b9d37cfa.png">

# How Has This Been Tested?

- Remove `${HOME}/xcash-official` directory;
- Run the GUI wallet and see it's working.

**Test Configuration**:
* Firmware version: XCASH Core 2.0.1 + MacOS Big Sur 11.2.3
* Hardware: MacBook Pro 13'' 2,4 GHz Quad-Core Intel Core i5, 8GB

# Checklist:

~- [x] I have added tests that prove my fix is effective or that my feature works~
~- [ ] New and existing unit tests pass locally with my changes~
~- [ ] Any dependent changes have been merged and published in downstream modules~
